docs/reference/hooks.md

```diff
@@ -0,0 +1,154 @@
+### Action file
+Schema of the Action file:
+
+|Property          |Description                                            |Required |Default Value
+|------------------|-------------------------------------------------------|---------|--------------------------------------|
+|name              |Identify the Action file                               |false    | If missing, filename is used instead 
+|on                |List of events that will trigger the hooks             |true     |
+|on<event>.branches|Glob pattern list of branches that triggers the hooks  |false    | If empty, Action runs on all branches
+|hooks             |List of hooks to be executed                           |true     | 
+|hook.id           |ID of the hook, must be unique within the `Action`     |true     | 
+|hook.type         |Type of the hook (currently only `webhook` supported   |true     | 
+|hook.properties   |Hook's specific configuration                          |true     | 
+
+Example:
+```yaml
+name: Good files check
+description: set of checks to verify that branch is good
+on:
+  pre-commit:
+  pre-merge:
+    branches:
+      - master
+hooks:
+  - id: no_temp
+    type: webhook
+    description: checking no temporary files found
+    properties:
+      url: "https://your.domain.io/webhook?notmp=true?t=1za2PbkZK1bd4prMuTDr6BeEQwWYcX2R"
+  - id: no_freeze
+    type: webhook
+    description: check production is not in dev freeze
+    properties:
+      url: "https://your.domain.io/webhook?nofreeze=true?t=1za2PbkZK1bd4prMuTDr6BeEQwWYcX2R"
+```
+
+## Run
+`Run` is the instantiation of the repo `Action` files when the event occurs.
+lakeFS will fetch, parse and filter the repository `Action` files and start to execute the `Hooks` under each `Action`.
+All executed `Hooks` (each with `hook_run_id`) exists in the context of that `Run` (`run_id`).   
+
+## How to upload Action files
+`Action` files should be uploaded with the prefix `_lakefs_actions/` to the lakeFS repository.
+When an actionable event (see Supported Events above) takes place, lakeFS will read all files with prefix `_lakefs_actions/`
+in the repository branch where the action occurred. 
+A failure to parse an `Action` file will result with the event aborted. 
+ 
+For example, lakeFS will search and execute all matching `Action` files with the prefix `lakefs://repo1/feature-1/_lakefs_actions/` on:
+1. Commit to `feature-1` branch on `repo1` repository. 
+2. Merge to `master` branch from `feature-1` branch on `repo1` repository.
+
+## Runs API
+OpenAPI endpoint exposes the results of `Runs` execution per repository, branch, commit and specific `Action`.
+The endpoint also allows to download the execution log of any executed `Hook` under each `Run` for observability.
+
+### Run results objectstore files 
+There are 2 types of files that are stored in the metadata section of lakeFS repository with each `Run`:
+1. `_lakefs/actions/log/<<runID>>/<<hookRunID>>.log` - Execution log of the specific `Hook` run.
```

Comment:
Why the double brackets? is that a markdown thing?

Reply:
Removed.